crates/fluvio/src/error.rs

```diff
@@ -45,6 +48,10 @@ To interact with this cluster, please install the matching CLI version using the
     ConsumerConfig(String),
     #[error("Encountered a runtime error in the user's SmartStream: {0}")]
     SmartStreamRuntime(#[from] SmartStreamRuntimeError),
+    #[error("Producer error")]
+    Producer(#[from] ProducerError),
+    #[error("An internal producer error occurred: {0}")]
+    InternalProducerError(String),
```

Comment:
There is nothing user can do.  Just mapped to Other instead of creating another variant